system/lvol.py

```diff
@@ -201,8 +213,22 @@ def main():
 
     lvs = parse_lvs(current_lvs)
 
+    if thinpool and lv:
+        for test_lv in lvs:
+            if test_lv['name'] == thinpool:
+                break
+        else:
+            module.fail_json(msg="Thin poll %s does not exist." % thinpool)
+
+    if thinpool and not lv:
+        target = thinpool
+    elif lv:
+        target = lv
+    else:
+        module.fail_json(msg="lv param is required unless thinpool param is specified.")
```

Comment:
Part of this code can be replaced with "required_one_of" ( see https://github.com/ansible/ansible-modules-core/blob/devel/cloud/digital_ocean/digital_ocean.py#L424 as a example )